### PR TITLE
DDF-5367 Update DefinitionParser to use Apache commons io

### PR DESCRIPTION
**Pull request by rzwiefel**
#### What does this PR do?
 - Instead of using felix fileinstaller use Apache Commons to watch for files. This means that file polling will only ever happen if and when the catalog-core-definitionparser bundle is started. 

#### Who is reviewing it? 
anyone

#### How should this be tested?
Ensure Definition files are installed reliably when starting the system. 

#### Any background context you want to provide?
Depending on certain startup timings, the file poller used in felix fileinstaller can start up and poll the definitions directory before the DefinitionsParser artifact installer was actually available to parse those files. This results in the files never being processed. 

#### What are the relevant tickets?
Fixes: #5367

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by rzwiefel** [comment]
build now

File: catalog/core/catalog-core-definitionparser/src/main/java/ddf/catalog/definition/impl/DefinitionParser.java [hunk 1]
```diff
@@ -138,6 +141,8 @@
           .setLenient()
           .create();
 
+  public static final int FILE_POLL_INTERVAL = 997;
```
Comment (jlcsmith) [inline catalog/core/catalog-core-definitionparser/src/main/java/ddf/catalog/definition/impl/DefinitionParser.java]: ✏️ could specify this is milliseconds in the name or in a comment

**Comment by jlcsmith** [comment]
Could you please update the commit / PR title with the issue #?

File: catalog/core/catalog-core-definitionparser/src/main/java/ddf/catalog/definition/impl/DefinitionParser.java (hunk 1 above)
Comment (rzwiefel) [reply catalog/core/catalog-core-definitionparser/src/main/java/ddf/catalog/definition/impl/DefinitionParser.java]: Done

**Comment by rzwiefel** [comment]
- Added onDestroy method to bean so the file watcher thread gets shut down when bundle stops/restarts

**Comment by jlcsmith** [comment]
build now

**Comment by jlcsmith** [comment]
Hero failed 
Bundle failed to install
```
09:56:18,536 | WARN  | -SNAPSHOT/deploy | ingServiceImpl$1ManagedPaxLoggingService  398 | he.felix.fileinstall | Error while starting bundle: file:/projects/ddf/distribution/ddf/target/ddf-2.19.0-SNAPSHOT/deploy/catalog-core-definitionparser-2.19.0-SNAPSHOT.jar
org.osgi.framework.BundleException: Could not resolve module: catalog-core-definitionparser [418]
  Unresolved requirement: Import-Package: ddf.catalog; version="[2.19.0,3.0.0)"

	at org.eclipse.osgi.container.Module.start(Module.java:444) ~[?:?]
	at org.eclipse.osgi.internal.framework.EquinoxBundle.start(EquinoxBundle.java:428) ~[?:?]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundle(DirectoryWatcher.java:1260) [16:org.apache.felix.fileinstall:3.6.4]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.startBundles(DirectoryWatcher.java:1233) [16:org.apache.felix.fileinstall:3.6.4]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.doProcess(DirectoryWatcher.java:520) [16:org.apache.felix.fileinstall:3.6.4]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.process(DirectoryWatcher.java:365) [16:org.apache.felix.fileinstall:3.6.4]
	at org.apache.felix.fileinstall.internal.DirectoryWatcher.run(DirectoryWatcher.java:316) [16:org.apache.felix.fileinstall:3.6.4]
```

**Comment by Bdthomson** [comment]
Hero in Progress with fresh m2

**Comment by Bdthomson** [comment]
build now

**Pull request merged**